docs/contributing/backend.md

```diff
@@ -0,0 +1,272 @@
+# Contributing to Nixopus Backend
+
+This guide provides detailed instructions for contributing to the Nixopus backend codebase.
+
+## Setup for Backend Development
+
+1. **Prerequisites**
+   - Go version 1.23.6 or newer
+   - PostgreSQL
+   - Docker and Docker Compose (recommended)
+
+2. **Environment Setup**
+
+   ```bash
+   # Clone the repository
+   git clone https://github.com/raghavyuva/nixopus.git
+   cd nixopus
+   
+   # Set up PostgreSQL database
+   createdb nixopus -U postgres
+   createdb nixopus_test -U postgres
+   
+   # Copy environment template
+   cp api/.env.sample api/.env
+   
+   # Install dependencies
+   cd api
+   go mod download
+   ```
+
+3. **Database Migrations**
```

Comment:
Database migrations are automatically performed when they run the application for the first time, so should not be a specific step to consider